src/algebra/group/pi.lean

```diff
@@ -137,6 +137,59 @@ variables {α β γ : Type*}
 
 end instance_lemmas
 
+section
+
+variables [decidable_eq I]
+
+/-- Extend a function from `{i}ᶜ : set I` to `I`. -/
+def extend_compl_single (i : I) (y : f i) (g : Π j : ({i}ᶜ : set I), f j) :
+  Π j : I, f j :=
+λ j, if h : j = i then eq.rec y h.symm else g ⟨j, h⟩
+
+@[simp] lemma extend_compl_single_same (i : I) (y : f i) (g : Π j : ({i}ᶜ : set I), f j) :
+  extend_compl_single i y g i = y :=
+dif_pos rfl
+
+@[simp] lemma extend_compl_single_of_noteq {i j : I} (h : j ≠ i) (y : f i)
+  (g : Π j : ({i}ᶜ : set I), f j) :
+  extend_compl_single i y g j = g ⟨j, h⟩ :=
+dif_neg h
+
+@[simp] lemma extend_compl_single_zero [Π i, has_zero (f i)] (i : I) (y : f i) :
+  extend_compl_single i y 0 = single i y :=
+rfl
+
+lemma forall_extend_compl_single (p : Π i, f i → Prop) {i : I} {y : f i}
+  {g : Π j : ({i}ᶜ : set I), f j} :
+  (∀ j, p j (extend_compl_single i y g j)) ↔ p i y ∧ ∀ j (hj : j ≠ i), p j (g ⟨j, hj⟩) :=
+(and_forall_ne i).symm.trans $ and_congr (by rw extend_compl_single_same) $
+  forall_congr $ λ j, forall_congr $ λ hj, by rw [extend_compl_single_of_noteq hj]
+
+@[simp] lemma extend_compl_single_sub_same [Π i, add_group (f i)] (i : I) (y₁ y₂ : f i)
```

Comment:
```suggestion
@[simp] lemma extend_compl_single_sub [Π i, add_group (f i)] (i : I) (y₁ y₂ : f i)
```
Can you instead prove this by showing `extend_compl_single_add`, then defining `extend_compl_single_add_hom`, then using `map_sub`?